pkgs/by-name/ch/chawan/package.nix

```diff
@@ -0,0 +1,62 @@
+{ lib
+, stdenv
+, fetchFromSourcehut
+, makeBinaryWrapper
+, curlMinimal
+, man
+, ncurses
+, nim
+, pandoc
+, perl
+, pkg-config
+, zlib
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "chawan";
+  version = "0-unstable-2024-03-01";
+
+  src = fetchFromSourcehut {
+    owner = "~bptato";
+    repo = "chawan";
+    rev = "87ba9a87be15abbe06837f1519cfb76f4bf759f3";
+    hash = "sha256-Xs+Mxe5/uoxPMf4FuelpO+bRJ1KdfASVI7rWqtboJZw=";
+    fetchSubmodules = true;
+  };
+
+  buildInputs = [ curlMinimal ncurses perl zlib ];
+  nativeBuildInputs = [
+    makeBinaryWrapper
+    nim
+    pandoc
+    pkg-config
+  ];
+
+  buildFlags = [ "all" "manpage" ];
+  installFlags = [
+    "DESTDIR=$(out)"
+    "PREFIX=/"
+  ];
+
+  postInstall =
+  let
+    makeWrapperArgs = ''
+      --set MANCHA_CHA $out/bin/cha \
+      --set MANCHA_MAN ${man}/bin/man \
+      --prefix MANPATH : $out/share/man:$(${man}/bin/man -w)
```

Comment:
Yeah I'm pretty sure it will, I'll try to work on a patch for it. mancha has some weird heuristics in this area so I'll probably send the patch upstream, thanks for pointing this out.

Reply:
Hi :)

I don't know what weird heuristics you ran into, but in any case, overriding MANPATH will probably not help.

mancha only ever touches MANPATH if -M is passed. In that case, it overrides it completely (like man(1) would). So it does not really make sense to set it specifically for the `man:` handler.

(In fact, mancha is just a wrapper around a modified version of w3mman2html.cgi, and it [looks like](https://github.com/NixOS/nixpkgs/blob/29d830243aeeded398237f0743f4d7265967df3c/pkgs/applications/networking/browsers/w3m/default.nix) w3m does not receive any MANPATH override in NixOS either.)

Reply:
Ah you're right, unsetting MANPATH actually fixes the problem. Now mancha can find system manuals on NixOS but it doesn't know where to find the manual for cha and mancha, I'll have to come up with a patch for that.

Thanks for the info! :)